🏃tests: add missing tests in repository github

cpanato opened the pull request:
**What this PR does / why we need it**:
In the `cmd/clusterctl/client/repository/repository_github_test.go` file was some TODO to implement missing tests, this PR adds those missing tests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

wfernandes commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,11 +30,204 @@ import (
+			},
+			want: &gitHubRepository{
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+
```
nit: remove extra whitespace

wfernandes commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,11 +30,204 @@ import (
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gitHub, err := newGitHubRepository(tt.field.providerConfig, tt.field.variableClient)
```
```suggestion
		t.Run(tt.name, func(t *testing.T) {
		        g := NewWithT(t)
			gitHub, err := newGitHubRepository(tt.field.providerConfig, tt.field.variableClient)
```
Move the `g := NewWithT(t)` from line 34 here to pass in the correct context to ginkgo.

Currently, the errors throw
```
    --- FAIL: Test_githubRepository_newGitHubRepository/can_create_a_new_GitHub_repo (0.00s)
        testing.go:864: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
```

wfernandes commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,11 +30,204 @@ import (
+}
+
+func Test_githubRepository_getComponentsPath(t *testing.T) {
+	g := NewWithT(t)
```
Same comment as above. Move this inside the `t.Run`

wfernandes commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,11 +30,204 @@ import (
+		t.Run(tt.name, func(t *testing.T) {
+			got := getComponentsPath(tt.path, tt.rootPath)
+
+			g.Expect(got).To(Equal(tt.want))
```
nit:
```suggestion
			g.Expect(getComponentsPath(tt.path, tt.rootPath)).To(Equal(tt.want))
```

wfernandes commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,11 +30,204 @@ import (
 
-//TODO: test GetFile
+func Test_githubRepository_getFile(t *testing.T) {
+	g := NewWithT(t)
```
move inside `t.Run`

wfernandes commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,11 +30,204 @@ import (
+
+	providerConfig := config.NewProvider("test", "https://github.com/o/r/releases/v0.4.1/file.yaml", clusterctlv1.CoreProviderType)
+
+	// test.NewFakeGitHu and handler for returning a fake release
```
nit: 
```suggestion
	// test.NewFakeGitHub and handler for returning a fake release
```

wfernandes commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,11 +30,204 @@ import (
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+				variableClient: nil,
+			},
+			want:    &gitHubRepository{},
```
I think in these cases we want a `nil`. I know we aren't necessarily asserting on this but as part of documentation for those reading the tests it would make sense to have this as `nil`.

cpanato commented:
thanks for your review @wfernandes, I'm worried that I've added all the `g:=NewWithT(t)` in the top of the tests 😅 for all the tests we converted to use gomega

PTAL

fabriziopandini commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,15 +30,203 @@ import (
+	providerConfig := config.NewProvider("test", "https://github.com/o/r/releases/v0.4.1/file.yaml", clusterctlv1.CoreProviderType)
+
+	// test.NewFakeGitHub and handler for returning a fake release
+	mux.HandleFunc("/repos/o/r/releases/tags/foo", func(w http.ResponseWriter, r *http.Request) {
```
should this be /v0.4.1?

cpanato replied:
yep :)

fabriziopandini commented on cmd/clusterctl/client/repository/repository_github_test.go:
```diff
@@ -30,15 +30,203 @@ import (
+	}{
+		{
+			name:     "Release and file exist",
+			release:  "foo",
```
v0.4.1?

cpanato replied:
done

wfernandes commented:
/lgtm
